### PR TITLE
Formatting for Display

### PR DESCRIPTION
**Pull request by emilylwan**
## Pull Request Summary

Parse the displayed text to get rid of voice name and fixed issues with not replacing -DOT- with .

### Change types

- [x] Non-breaking change

### How to manually test this PR (QA instructions)
Upload a transcript and check that the text formatting has periods and that voice name is replaced with just the name.

### CODE REVIEW

File: back-end/helpers/intent_analyzer.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ export default class IntentAnalyzer{
     }
 
     const intents = {
-      question: model.question.replace("-DOT-", "."),
+      question: model.question.replaceAll("-DOT-", "."),
```
Comment (WeissSa) [inline back-end/helpers/intent_analyzer.js:20]: good catch. Maybe add a test so this doesn't happen again in future (

File: back-end/helpers/intent_analyzer.js [hunk 2]
```diff
@@ -27,7 +27,7 @@ export default class IntentAnalyzer{
   replacePeriods(map) {
     const newMap = {};
     Object.entries(map).forEach(([intent, percentage]) => {
-        const replacedIntent = intent.replace("-DOT-", ".")
+        const replacedIntent = intent.replaceAll("-DOT-", ".")
```
Comment (WeissSa) [inline back-end/helpers/intent_analyzer.js:30]: +1

File: back-end/helpers/transcript_data_formatter.js [hunk 3]
```diff
@@ -17,7 +18,14 @@ export default class TranscriptFormatter {
...
             const id = entry["project_id"];
-            const message = tracePayload.payload.message;
+            let message = tracePayload.payload.message;
+            if (message.includes("\"")) {
+                // Regex: if it matches " and anything in between the next " is copied
+                const name = message.match(/"([^)]+)"/)[1] + ": ";
+                // Regex: Match < and any text until > then replace with an empty string
+                message = message.replaceAll(/<.*?>/g, '');
+                message = name + message;
+            }
```
Comment (WeissSa) [inline back-end/helpers/transcript_data_formatter.js:28]: Maybe add this to the description of a test so it is explicit

**Review by WeissSa — CHANGES_REQUESTED** [review]
Small testing things (as is my usual), but overall really good

**Review by WeissSa — APPROVED** [review]
Awesome :sunglasses:

**Review by helias22 — COMMENTED** [review]
Great catch on the multiple dots problem. I also saw that today in the presentation and I wanted to add another ticket to solve it :)
About getting rid of the voice I don't think It's completely clean yet. This is what I get when I try it with my sample transcript.
Screen Shot 2022-11-17 at 8 46 59 PM

**Comment by emilylwan** [comment]
@helias22 I found the problem and hopefully fixed it. Let me know if it works now. Though also on another note, are there going to be quotations around the main message in transcripts? From what I can see on the data Denys sent us, he didn't put quotes around the text.

**Review by helias22 — APPROVED** [review]
Yes the problem is solved. About the quotations, I think I accidentally added those when I was adding to the transcript. Great catch :))

**Review by williamyao27 — APPROVED** [review]
LGTM!

**Pull request merged**